### ISSUE TITLE
[Feature Request] Lifted vjp3 in NNX

### ISSUE THREAD

**Issue by qGentry** [google/flax#5114]
Hi guys, it would be really nice if Flax.NNX API had lifted `jax._src.api.vjp3` under `nnx.vjp3` API. 

We found vjp3 to be very beneficial in terms of memory usage when training models with scan over layers and gradient accumulation because it avoid doubling memory usage for intermediate gradients thanks to the new JAX 'ref' semantics.

**Comment by vfdev-5** [comment]
Currently, nnx goal is to provide the support for native jax transformations instead of writing and maintaining wrappers (like `nnx.jit` etc). It is unlikely, flax will provide new transformations.
On the other hand, vjp3 is a default code path in jax: 
https://github.com/jax-ml/jax/blob/5fae04665f6d711de9f242743072d783febe157b/jax/_src/api.py#L2232-L2233
and 
```python
import jax
print(jax.config.jax_vjp3)
# True
```

I also think `jax.vjp` should work with `nnx.Module` etc.

**Comment by qGentry** [comment]
Ah, I didn't know that jax already uses vjp3 as a default backend for grad. 
Closing this issue then, thanks!
